build.gradle

```diff
@@ -227,30 +233,51 @@ subprojects {
             java {
                 srcDirs += generatedDir
             }
-            output.classesDir = sourceSets.main.output.classesDir
         }
         main {
+            compileClasspath += files([sourceSets.generated.java.outputDir])
+            runtimeClasspath += files([sourceSets.generated.java.outputDir])
             java {
-                compileClasspath += generated.output
-                runtimeClasspath += generated.output
-            }
-            scoverage {
-                compileClasspath += generated.output
-                runtimeClasspath += generated.output
+                exclude '**/*.jaxb'
+                exclude '**/README'
             }
         }
         test {
-            java {
-                compileClasspath += generated.output
-                runtimeClasspath += generated.output
-            }
-            testScoverage {
-                compileClasspath += generated.output
-                runtimeClasspath += generated.output
-            }
+            compileClasspath += configurations.compileOnly
+            runtimeClasspath += configurations.compileOnly
+            compileClasspath += files([sourceSets.generated.java.outputDir])
+            runtimeClasspath += files([sourceSets.generated.java.outputDir])
+        }
+        integrationTest {
+            compileClasspath += configurations.compileOnly
+            runtimeClasspath += configurations.compileOnly
+            compileClasspath += files([sourceSets.generated.java.outputDir])
+            runtimeClasspath += files([sourceSets.generated.java.outputDir])
+        }
+        scoverage {
+            compileClasspath += configurations.compileOnly
+            runtimeClasspath += configurations.compileOnly
+            compileClasspath += files([sourceSets.generated.java.outputDir, sourceSets.main.java.outputDir])
+            runtimeClasspath += files([sourceSets.generated.java.outputDir, sourceSets.main.java.outputDir, sourceSets.test.groovy.outputDir])
+        }
+        testScoverage {
+            compileClasspath += configurations.compileOnly
+            runtimeClasspath += configurations.compileOnly
+            compileClasspath += files([sourceSets.generated.java.outputDir, sourceSets.main.java.outputDir])
+            runtimeClasspath += files([sourceSets.generated.java.outputDir,
+                                       sourceSets.main.java.outputDir,
+                                       sourceSets.main.groovy.outputDir,
+                                       sourceSets.test.groovy.outputDir,
+                                       file("${project.buildDir}/resources/scoverage/")
+            ])
```

Comment:
Jeez, that got nasty.

Reply:
That was the one that was causing all the squirreliness yesterday.